goon/browserassets/js/browserOutput.js

```diff
@@ -171,25 +171,67 @@ function setHighlightColor(match) {
 	match.style.background = opts.highlightColor
 }
 
+function escapeRegex(input){ // put this in a function incase it ever needs to be used elsewhere, makes code cleaner
+	return input.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+}
+
 //Highlights words based on user settings
 function highlightTerms(el) {
 	if(regexHasError) return; //just stop right there ig the regex is gonna except
+	function highlightRecursor(element,term){ //recursor function to do the highlighting proper
+		var regex = new RegExp("("+term+")","gi")
+		function replace(str) {
+			return str.replace(regex,'<span style="background-color:'+opts.highlightColor+'">$&</span>')
+		}
+		var s=''
+		var work=element.innerHTML;
+		var work_lc=work.toLowerCase()
+		var ind=0;
+		while(ind < work_lc.length) {
+			console.log(s);
+			var next_term=work_lc.substring(ind).search(regex);
+            if(next_term!=-1)next_term+=ind;
+			var next_tag =work_lc.indexOf('<',ind)
+			if(next_tag == -1) {
+				s+=replace(work.substring(ind));
+				break;
+			}
+			else if(next_term==-1) {
+				s+=work.substring(ind)
+				break;
+			}
+			else if(next_tag < next_term) {
+				var temp=work_lc.indexOf('>',next_tag)
+				s+=work.substring(ind,temp+1);
+				ind=temp+1;
+			}
+			else {
+				s+=replace(work.substring(ind,next_tag));
+				ind=next_tag
+			}
+		}
+		
+		element.innerHTML=s;
+	}
+	
 	for (var i = 0; i < opts.highlightTerms.length; i++) { //Each highlight term
 		if(opts.highlightTerms[i]) {
 			if(!opts.highlightRegexEnable){
-				if(el.innerText.toString().toLowerCase().includes(opts.highlightTerms[i].toLowerCase())) //match normally
-				el.innerHTML = '<span style="background-color:'+opts.highlightColor+'">'+el.innerHTML+'</span>' //encloseincludes
-				continue;
+				var innerTerms = opts.highlightTerms[i].split(" ")
+				for(var a in innerTerms){
+					highlightRecursor(el,escapeRegex(innerTerms[a]))
+				}
 			}
-			var rexp;
-			try{
-				rexp = new RegExp(opts.highlightTerms[i],"gmi")
-			} catch(e){
-				el.innerHTML+='<br/><span style="boldwarning"> Your highlight regex - '+opts.highlightTerms[i]+' - is malformed. Thrown exception: '+e+'</span>'
-				regexHasError = true;
-				return;
+			else {
+				try{
+					new RegExp(opts.highlightTerms[i],"gmi") // check to make sure the pattern wont cause issues
+				} catch(e){
+					el.innerHTML+='<br/><span style="color:#000;background-color:#FFFF00;" class="bold"> Your highlight regex pattern - '+opts.highlightTerms[i]+' - is malformed.<br/>Your highlights have been disabled until they are next edited<br/>Thrown exception: '+e+'</span>'
```

Comment:
The user should be getting the error when they are editing the highlights, not when they are receiving a message.

Reply:
Your highlight is invalid x10000

Reply:
are you going to actually do anything about it
it looks like around 1107 would be a good place to do that actual check

Reply:
The `internalOutput` at 1108 causes this check to be done already, however if a saved invalid regex is in options at the time the chat loads checking only in edit wouldnt catch that and cause the chat panel to fail to load.